### PR TITLE
Feature/posthog

### PR DESCRIPTION
**Pull request by kektobiologist**

### CODE REVIEW

File: extension/src/content/RPCs/actions.ts [hunk 1]
```diff
@@ -89,7 +90,7 @@ const scrollElementIntoView = async (element: Element) => {
   }
 };
 
-const dropText = async (textToDrop: string, element: Element) => {
+export const dropText = async (textToDrop: string, element: Element) => {
```
Comment (ppsreejith) [inline extension/src/content/RPCs/actions.ts:93]: Should this be exported?

File: extension/src/content/RPCs/actions.ts [hunk 2]
```diff
@@ -108,7 +109,9 @@ export const uClick = async (selector: QuerySelector, index: number = 0) => {
     return await user.click(element)
   }
 }
-
+if (process.env.NODE_ENV == 'development' || true) {
```
Comment (ppsreejith) [inline extension/src/content/RPCs/actions.ts]: Can we use `configs.IS_DEV` here instead?

File: apps/src/posthog/inject.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { get, isEmpty } from "lodash"
```
Comment (kektobiologist) [inline apps/src/posthog/inject.ts:1]: @ppsreejith check this file if its fine

File: extension/src/content/RPCs/fetchData.ts [hunk 4]
```diff
@@ -1,19 +1,44 @@
...
+    headerKey: string
+  }
+): Promise<T> {
+  if (csrfInfo && csrfInfo?.cookieKey && csrfInfo?.headerKey) {
+    const token = getCookie(csrfInfo.cookieKey);
+    if (token) {
+      headers[csrfInfo.headerKey] = String(token);
+    }
+  }
+  let options: RequestInit = {
```
Comment (ppsreejith) [inline extension/src/content/RPCs/fetchData.ts]: Can this be `const` ?

**Review by ppsreejith — CHANGES_REQUESTED** [review]

File: extension/src/content/debug.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import userEvent, { PointerEventsCheckLevel } from '@testing-library/user-event';
+import { fireEvent } from '@testing-library/dom';
+import { dropText } from './RPCs/actions'
```
Comment (ppsreejith) [inline extension/src/content/debug.ts]: Maybe we can not use `dropText` here?

File: extension/src/content/index.ts [hunk 6]
```diff
@@ -25,6 +25,10 @@ async function _init(localConfigs: Promise<object>) {
   if (inject) {
     setupScript(`${tool}.bundle.js`)
   }
+  if (configs.IS_DEV) {
+    console.log('Injecting debug script')
+    setupScript(`debug.bundle.js`)
+  }
   if (!configs.IS_DEV) {
```
Comment (ppsreejith) [inline extension/src/content/index.ts:32]: Use an `else` here? :P

File: extension/src/content/index.ts [hunk 7]
```diff
@@ -36,6 +40,7 @@ async function _init(localConfigs: Promise<object>) {
     initWindowListener(posthogRPCs)
   }
 
+  setupScript('debug.bundle.js')
```
Comment (ppsreejith) [inline extension/src/content/index.ts]: Should this be removed?

File: web/src/app/rpc.ts [hunk 8]
```diff
@@ -43,7 +43,7 @@ export const setMinusxMode = (mode: string) => sendMessage("setMinusxMode", [mod
 export const toggleMinusXRoot = (className: ValidMinusxRootClass, value?: boolean) => sendMessage("toggleMinusXRoot", [className, value], {log_rpc: true})
 export const captureVisibleTab = () => sendMessage("captureVisibleTab", [], {log_rpc: true})
 export const getElementScreenCapture = (selector: QuerySelector) => sendMessage("getElementScreenCapture", [selector], {log_rpc: true})
-export const uSelectAllText = (shouldDelete = false) => sendMessage("uSelectAllText", [shouldDelete], {log_rpc: true})
+export const uSelectAllText = (shouldDelete = false, deleteOps = ['delete']) => sendMessage("uSelectAllText", [shouldDelete, deleteOps], {log_rpc: true})
```
Comment (ppsreejith) [inline web/src/app/rpc.ts]: Remove deleteOps?

**Review by ppsreejith — CHANGES_REQUESTED** [review]

File: apps/src/posthog/inject.ts [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    }
+    return get(appContext, path)
+  }
+  return null
+}
+
+// HACK: resize hogQL editor so that normal queries can fit
+let interval = setInterval(() => {
+  const hogQLContainerSelector = querySelectorMap["hoql_container_to_resize"]
+  const container = document.evaluate(hogQLContainerSelector.selector, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE).singleNodeValue as HTMLElement
```
Comment (ppsreejith) [inline apps/src/posthog/inject.ts]: Can you use the pre-existing selector for this?

**Comment by ppsreejith** [comment]
Lgtm! Let's ship!

**Pull request merged**